framework/flamingo/sessions.go

```diff
@@ -73,7 +75,12 @@ func (m *SessionModule) Inject(config *struct {
 	m.storeLength = int(config.StoreLength)
 	m.maxAge = int(config.MaxAge)
 	m.path = config.Path
-	m.redisHost, m.redisPassword, m.redisDatabase = getRedisConnectionInformation(config.RedisURL, config.RedisHost, config.RedisPassword, config.RedisDatabase)
+
+	m.redisHost = getRedisHost(config.RedisURL, config.RedisHost)
+	m.redisUsername = getRedisUsername(config.RedisURL, config.RedisUsername)
+	m.redisPassword = getRedisPassword(config.RedisURL, config.RedisPassword)
+	m.redisDatabase = getRedisDatabase(config.RedisURL, config.RedisDatabase)
```

Comment:
With this function split, we parse the URL 4 times. No idea if it is an expensive operation

Reply:
yeah, moved parsing out